jupyterhub/values.yaml

```diff
@@ -143,6 +143,16 @@ auth:
 
 singleuser:
   extraTolerations: []
+  nodeSelector: {}
+  extraNodeAffinity:
```

Comment:
Instead of exposing six different configurables, can we expose the Kubernetes `.affinity` structure directly? I think that would be a much better and better-documented experience.

Reply:
We want the affinity to be utilized by the image puller as well, which is configured directly from a heml template. But we only want the daemonset to use the affinity **partially**, for example the daemonset should not care about `affinity.podAffinity` or but should care about `affinity.nodeAffinity.requiredDuring...`.

Attempting to extract pieces from a big dict passed to a value in config.yaml is not plausible even though I can imagine someone have made some super complicated trick to accomplish.

Reply:
Then doing so would also rule out having a configurable preset of `require` `prefer` `ignore` matching node affinity for user / core pods with user / core pools for example.

Reply:
We're talking about one level of a dict `affinity.nodeAffinity`, aren't we? Isn't that less complex than what's happening here, where we are mapping several levels of a dict to different things?